MSAL/src/native_auth/controllers/sign_up/MSALNativeAuthSignUpControlling.swift

```diff
@@ -26,16 +26,15 @@
 
 protocol MSALNativeAuthSignUpControlling: AnyObject {
 
-    typealias SignUpStartPasswordControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpPasswordStartResult>
-    typealias SignUpStartCodeControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpStartResult>
+    typealias SignUpStartControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpStartResult>
     typealias SignUpResendCodeControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpResendCodeResult>
     typealias SignUpSubmitCodeControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpVerifyCodeResult>
     typealias SignUpSubmitPasswordControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpPasswordRequiredResult>
     typealias SignUpSubmitAttributesControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpAttributesRequiredResult>
 
-    func signUpStartPassword(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartPasswordControllerResponse
+    func signUpStartPassword(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartControllerResponse
 
-    func signUpStartCode(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartCodeControllerResponse
+    func signUpStartCode(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartControllerResponse
```

Comment:
I leave the same [comment that I left in the abandoned PR](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1935#discussion_r1422913042) to keep track of it.

For clarification, I wasn't asking for a full refactor of the controller, just its interface. I see that @nilo-ms has [commented](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1942#discussion_r1427780228) something similar for `SignInController`.

Reply:
Just completed a full refactoring :)